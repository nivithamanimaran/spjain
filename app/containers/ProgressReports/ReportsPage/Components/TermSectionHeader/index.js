import React from 'react';
import { Box, Grid, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { useTranslation } from 'react-i18next';
import KenSwitch from '../../../../../components/KenSwitch/index';

const useStyles = makeStyles(theme => ({
  termLabel: {
    color: theme.palette.KenColors.gradeTermLabel,
    fontSize: 12,
    [theme.breakpoints.down('xs')]: {
      fontSize: 12,
    },
  },
  termWrap: {
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingTop: 16,
    paddingBottom: 12,
    // paddingLeft: 18,
    // paddingRight: 32,
  },
  subHeader: {
    display: 'flex',
    alignItems: 'center',
    [theme.breakpoints.down('xs')]: {
      marginTop: 10,
      marginLeft: 10,
    },
  },
  switchLabel: {
    color: theme.palette.KenColors.gradeSectionHeaderLabel,
    fontSize: 13,
    marginLeft: 8,
  },
}));

export default function TermSectionHeader(props) {
  const classes = useStyles();
  const { termLabel, onShowSwitch = true } = props;
  const { t } = useTranslation();

  const handleChange = e => {
    console.log(e.target.checked, 'e...');
  };

  return (
    <Grid
      container
      className={classes.termWrap}
      data-testid="term-section-header"
    >
      <Grid className={classes.termLabel} data-testid="term-section-title">
        {termLabel}
      </Grid>
      {onShowSwitch && (
        <Box className={classes.subHeader}>
          <KenSwitch
            onChangeSwitch={handleChange}
            switchLabel={
              <Typography className={classes.switchLabel}>
                {t('labels:View_Table')}
              </Typography>
            }
          />
        </Box>
      )}
    </Grid>
  );
}
