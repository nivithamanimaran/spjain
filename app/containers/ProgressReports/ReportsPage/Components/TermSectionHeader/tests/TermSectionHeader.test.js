import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import ReactDOM from 'react-dom';
import TermSectionHeader from '../index';
import ThemeProvider from '../../../../../../utils/themeProvider';
import { BrowserRouter as Router } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

describe('term-Section-Header', () => {
  const div = document.createElement('div');
  it('Term Section Header renders correctly without props', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <TermSectionHeader />
        </Router>
      </ThemeProvider>,
      div
    );

    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <TermSectionHeader />
        </Router>
      </ThemeProvider>
    );

    expect(getByTestId('term-section-header')).toMatchSnapshot();
  });

  it('Term Section Header renders correctly with props', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <GradeSectionHeader termLabel="title" />
        </Router>
      </ThemeProvider>,
      div
    );

    const termLabel = queryByTestId('term-section-title');
    expect(termLabel).toBeTruthy();
    expect(termLabel.textContent).toBe('Title');
  });
});
