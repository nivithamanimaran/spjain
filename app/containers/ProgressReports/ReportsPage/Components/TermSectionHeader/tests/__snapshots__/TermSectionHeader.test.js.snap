// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`term-Section-Header Term Section Header renders correctly without props 1`] = `
<div
  class="MuiGrid-root makeStyles-termWrap-2 MuiGrid-container"
  data-testid="term-section-header"
>
  <div
    class="MuiGrid-root makeStyles-termLabel-1"
    data-testid="term-section-title"
  />
  <div
    class="MuiBox-root MuiBox-root-16 makeStyles-subHeader-3"
  >
    <div
      class="MuiFormGroup-root"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          class="MuiSwitch-root Component-root-6"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-12 MuiSwitch-switchBase Component-switchBase-7 MuiSwitch-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-15 MuiSwitch-input"
                name="checked"
                type="checkbox"
                value=""
              />
              <span
                class="MuiSwitch-thumb Component-thumb-8"
              />
            </span>
          </span>
          <span
            class="MuiSwitch-track Component-track-9"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          <p
            class="MuiTypography-root makeStyles-switchLabel-4 MuiTypography-body1"
          >
            labels:View_Table
          </p>
        </span>
      </label>
    </div>
  </div>
</div>
`;
